.golangci.yaml

```diff
@@ -0,0 +1,216 @@
+  dupl:
+    # tokens count to trigger issue, 150 by default
+    threshold: 150
+  goconst:
+    # minimal length of string constant, 3 by default
+    min-len: 3
+    # minimal occurrences count to trigger, 3 by default
+    min-occurrences: 3
+  depguard:
+    list-type: blacklist
+    include-go-root: false
+    packages:
+    # List of packages that we would want to blacklist for... reasons.
+  misspell:
+    # Correct spellings using locale preferences for US or UK.
+    # Default is to use a neutral variety of English.
+    # Setting locale to US will correct the British spelling of 'colour' to 'color'.
+    locale: US
+  lll:
+    # max line length, lines longer will be reported. Default is 120.
+    # '\t' is counted as 1 character by default, and can be changed with the tab-width option
+    line-length: 150
+    # tab width in spaces. Default to 1.
+    tab-width: 1
+  unused:
+    # treat code as a program (not a library) and report unused exported identifiers; default is false.
+    # XXX: if you enable this setting, unused will report a lot of false-positives in text editors:
+    # if it's called for subdir of a project it can't find funcs usages. All text editor integrations
+    # with golangci-lint call it on a directory with the changed file.
+    check-exported: false
+  unparam:
+    # call graph construction algorithm (cha, rta). In general, use cha for libraries,
+    # and rta for programs with main packages. Default is cha.
+    algo: cha
+
+    # Inspect exported functions, default is false. Set to true if no external program/library imports your code.
+    # XXX: if you enable this setting, unparam will report a lot of false-positives in text editors:
+    # if it's called for subdir of a project it can't find external interfaces. All text editor integrations
+    # with golangci-lint call it on a directory with the changed file.
+    check-exported: false
+  nakedret:
+    # make an issue if func has more lines of code than this setting and it has naked returns; default is 30
+    max-func-lines: 20
+  prealloc:
+    # XXX: we don't recommend using this linter before doing performance profiling.
+    # For most programs usage of prealloc will be a premature optimization.
+
+    # Report preallocation suggestions only on simple loops that have no returns/breaks/continues/gotos in them.
+    # True by default.
+    simple: true
+    range-loops: true # Report preallocation suggestions on range loops, true by default
+    for-loops: false # Report preallocation suggestions on for loops, false by default
+  gocritic:
+    # which checks should be enabled; can't be combined with 'disabled-checks';
+    # default are: [appendAssign assignOp caseOrder dupArg dupBranchBody dupCase flagDeref
+    # ifElseChain regexpMust singleCaseSwitch sloppyLen switchTrue typeSwitchVar underef
+    # unlambda unslice rangeValCopy defaultCaseOrder];
+    # all checks list: https://github.com/go-critic/checkers
+    enabled-checks:
```

Comment:
Should we instead use `disabled-checks`, so when new ones are added, we don't have to track it, we just start seeing failures?

I mean, I understand that that sort of "unpinning" mentality is not a prod mentality, but seems like it would be for us like a newsfeed for new checks :)

Reply:
Not IMHO. As you mention it yourself that would go against "prod mentality" or even general best-practices. Config files, etc are there to control your build environment, you don't want random things to change and your linting to fail just because your package manager happened to have upgraded `golangci-lint`.

Reply:
Not particularly convinced, but def do as you please.

I guess getting the odd new check in our face right away when _we_ update the ci tool seems dandy to me!
https://github.com/42wim/matterbridge/blob/a8dddf88ca27d60ed282d71f04d085d68a8ef674/.travis.yml#L31

One less tedious bit of list cross-checking in the world? Yes, please! Especially when it's a tool we're already expecting to say "Hey, look at this thing you weren't expecting, doofus!" :)

Reply:
Oh, hey, speaking of which: Let's pull GOLANGCI_VERSION into global envvar hash

Reply:
Done for the `GOLANGCI_VERSION`.